db/migrate/20190911050155_alter_build_indexes.rb

```diff
@@ -0,0 +1,9 @@
+class AlterBuildIndexes < ActiveRecord::Migration[5.2]
+  def change
+    remove_index :builds, [:git_sha, :dockerfile]
+    remove_index :builds, [:git_sha, :image_name]
+
+    add_index :builds, [:git_sha, :dockerfile, :external_url], unique: true, length: {git_sha: 80, dockerfile: 80, external_url: 255}
```

Comment:
I don't think this works since and index with null in it is never matching ... so this allows unlimited git_sha-X/dockerfile-Y/nil rows

Reply:
that's what I want.